content/5_Plat/Sqrt.mdx

```diff
@@ -46,62 +46,153 @@ $x$ and the element we want to change it to.
-	blocks[x / block_sz] -= nums[x];
+/** O(1) update to set nums[x] to v */
+void update(int x, int v) { 
+	blocks[x / block_size] -= nums[x];
 	nums[x] = v;
-	blocks[x / block_sz] += nums[x];
+	blocks[x / block_size] += nums[x];
 }
 
-ll query(int r) {
-	ll ret = 0;
-	for(int i = 0; i < r / block_sz; i++) ret += blocks[i];
-	for(int i = r / block_sz * block_sz; i < r; i++) ret += nums[i];
-	return ret;
+/** O(sqrt(n)) query for sum of [0, r) */
+long long query(int r) {
+	long long res = 0;
+	for (int i = 0; i < r / block_size; i++) {
+		res += blocks[i];
+	}
+	for (int i = (r / block_size) * block_size; i < r; i++) {
+		res += nums[i];
+	}
+	return res;
 }
 
-ll query(int l, int r) {
+/** O(sqrt(n)) query for sum of [l, r) */
+long long query(int l, int r) {
 	return query(r) - query(l - 1);
 }
 
 int main() {
-	setIO();
+	int n, q; 
+	cin >> n >> q;
+	block_size = (int) ceil(sqrt(n));
+	blocks.assign(block_size, 0);
+	nums.assign(n, 0);
+
+	for (int i = 0; i < n; i++) {
+		cin >> nums[i];
+	}
+	for (int i = 0; i < n; i++) {
+		blocks[i / block_size] += nums[i];
+	}
+	for (int i = 0; i < q; i++) {
+		int t, l, r; 
+		cin >> t >> l >> r;
+		if (t == 1) {
+			update(l - 1, r);
+		} else {
+			cout << query(l, r) << "\n";
+		}
+	}
+}
+```
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class DRSQ {
+	static int blockSize;
+	static long[] blocks;
+	static int[] nums;
+
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter pw = new PrintWriter(System.out);
+		StringTokenizer st = new StringTokenizer(br.readLine());
 
-	int n, q; cin >> n >> q;
-	block_sz = (int) sqrt(n);
+		int n = Integer.parseInt(st.nextToken());
+		int q = Integer.parseInt(st.nextToken());
+		blockSize = (int) Math.ceil(Math.sqrt(n));
 
-	for(int i = 0; i < n; ++i) cin >> nums[i];
-	for(int i = 0; i < n; ++i) blocks[i / block_sz] += nums[i];
-	for(int i = 0; i < q; ++i) {
-		int t, l, r; cin >> t >> l >> r;
-		if (t == 1) upd(l - 1, r);
-		else cout << query(l, r) << "\n";
+		nums = Arrays.stream(br.readLine().split(" "))
+			.mapToInt(Integer::parseInt)
+			.toArray();
+		blocks = new long[blockSize];
+		for (int i = 0; i < n; i++) {
+			blocks[i / blockSize] += nums[i];
+		}
+
+		for (int i = 0; i < q; i++) {
+			st = new StringTokenizer(br.readLine());
+			int type = Integer.parseInt(st.nextToken());
+			int l = Integer.parseInt(st.nextToken());
+			int r = Integer.parseInt(st.nextToken());
+
+			if (type == 1) {
+				update(l - 1, r);
+			} else {
+				pw.println(query(l, r));
+			}
+		}
+
+		br.close();
+		pw.close();
+	}
+
+	/** O(1) update to set nums[x] to v */
+	static void update(int x, int v) {
+		blocks[x / blockSize] -= nums[x];
+		nums[x] = v;
+		blocks[x / blockSize] += nums[x];
+	}
+
+	/** O(sqrt(n)) query for sum of [l, r) */
+	static long query(int l, int r) {
+		return query(r) - query(l - 1); 
```

Comment:
for data structure queries like this, i'd appreciate if you made this a class (likewise for c++)